13/index.js

```diff
@@ -0,0 +1,251 @@
+                model.cursor[0] += x;
+                if (model.cursor[0] < 0)
+                    model.cursor[0] += 3;
+                else if (model.cursor[0] > 2)
+                    model.cursor[0] -= 3;
+                model.cursor[1] += y;
+                if (model.cursor[1] < 0)
+                    model.cursor[1] += 3;
+                else if (model.cursor[1] > 2)
+                    model.cursor[1] -= 3;
+                view.update_view(model.field, model.cursor);
+                model.save_game_state();
+            }
+        },
+        confirm_move: function () {
+            if (model.state === 0 || model.state === 1) {
+                var cursor_index = 3 * model.cursor[1] + model.cursor[0];
+                if (model.field[cursor_index] === 0) {
+                    var pending_state = model.state === 0 ? 1 : 0;
+                    var pending_value = model.state === 0 ? 1 : 2;
+                    model.state = 2;
+                    setTimeout(function () {
+                        model.field[cursor_index] = pending_value;
+                        var result = game_result(model.field);
+                        if (result === 1) {
+                            view.update_header("Player O won!");
+                            model.state = 3;
+                        } else if (result === 2) {
+                            view.update_header("Player X won!");
+                            model.state = 3;
+                        } else if (!some(model.field, is_field_empty)) {
+                            view.update_header("Draw");
+                            model.state = 3;
+                        } else {
+                            view.update_header("Player " + (pending_state === 0 ? "O" : "X") + " moves");
+                            model.state = pending_state;
+                        }
+                        model.save_game_state();
+                        view.update_view(model.field, model.cursor);
+                    }, 300);
+                }
+            }
+        },
+        reset: function () {
+            this.model.field = [0, 0, 0, 0, 0, 0, 0, 0, 0];
+            this.model.cursor = [1, 1];
+            this.model.state = 0;
+            this.model.save_game_state();
+            this.view.update_view(model.field, model.cursor);
+            this.view.update_header("Player " + (model.state === 0 ? "O" : "X") + " starts game");
+        },
+        any_key: function () {
+            if (model.state === 3) {
+                controller.reset();
+                model.save_game_state();
+            }
+        }
+    }
+    if(model.has_saved_game_state()) {
+        model.load_game_state();
+        view.update_view(controller.model.field, controller.model.cursor);
+    }
+    else
+        controller.reset();
+    return controller;
+}
+
+function is_field_empty(v) {
+    return v === 0;
+}
+
+function some(a1, f1) {
+    for (var i = 0; i < a1.length; i++)
```

Comment:
There is `Array.prototype.some` method. You can use it. [Here ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some?v=control) you can find documentation